packages/imready/src/reactive.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  observe,
+  reactive,
+  ReactiveAdapter,
+  ReactiveObject,
+} from "@cfcs/core";
+import { EVENTS, METHODS } from "./consts";
+import ImReady from "./ImReady";
+import {
+  ArrayFormat,
+  ImReadyEvents,
+  ImReadyMethods,
+  ImReadyReactiveProps,
+  ImReadyReactiveState,
+} from "./types";
+import { toArray } from "./utils";
+
+export interface ImReadyData {
+  children: HTMLElement[];
+  props: Partial<ImReadyReactiveProps>;
+}
+
+export type ReactiveImReady = ReactiveObject<{
+  imReady: ImReady;
+  preReadyCount: number;
+  readyCount: number;
+  errorCount: number;
+  totalErrorCount: number;
+  totalCount: number;
+  isPreReady: boolean;
+  isReady: boolean;
+  hasError: boolean;
+  isPreReadyOver: boolean;
+  check(elements: ArrayFormat<HTMLElement>);
+  getTotalCount();
+  clear();
+}>;
+
+export const REACTIVE_IMREADY: ReactiveAdapter<
+  ReactiveImReady,
+  ImReadyReactiveState,
+  keyof ImReadyMethods,
+  ImReadyData,
+  ImReadyEvents
+> = {
+  methods: METHODS,
+  events: EVENTS,
+  state: {
+    preReadyCount: 0,
+    readyCount: 0,
+    errorCount: 0,
+    totalErrorCount: 0,
+    totalCount: 0,
+    isPreReady: false,
+    isReady: false,
+    hasError: false,
+    isPreReadyOver: false,
+  },
+  created(data) {
+    const imReady = new ImReady(data.props);
+    const {
+      useError,
+      usePreReadyElement,
+      useReadyElement,
+      usePreReady,
+      useReady,
+    } = data.props;
+    const preReadyCount = observe(0);
+    const readyCount = observe(0);
+    const errorCount = observe(0);
+    const totalCount = observe(0);
+    const totalErrorCount = observe(0);
+    const isPreReady = observe(false);
+    const isReady = observe(false);
+    const hasError = observe(false);
+    const isPreReadyOver = observe(false);
+
+    imReady
+      .on("error", (e) => {
+        if (useError) {
+          hasError.current = true;
+          errorCount.current = e.errorCount;
+          totalErrorCount.current = e.totalErrorCount;
+        }
+      })
+      .on("preReadyElement", (e) => {
+        if (usePreReadyElement) {
+          preReadyCount.current = e.preReadyCount;
+        }
+      })
+      .on("readyElement", (e) => {
+        if (useReadyElement) {
+          readyCount.current = e.readyCount;
+          isPreReadyOver.current = e.isPreReadyOver;
+        }
+      })
+      .on("preReady", () => {
+        if (usePreReady) {
+          isPreReady.current = true;
+        }
+      })
+      .on("ready", () => {
+        if (useReady) {
+          isReady.current = true;
+        }
+      });
+    return reactive({
+      imReady: imReady,
+      preReadyCount,
+      readyCount,
+      errorCount,
+      totalErrorCount,
+      totalCount,
+      isPreReady,
+      isReady,
+      hasError,
+      isPreReadyOver,
+      check: imReady.check,
```

Comment:
Since this is a method, `this` changes.

(...args: any[]) => imready.....(...args);